src/main/java/io/pivotal/migration/MigrationClient.java

```diff
@@ -745,6 +746,13 @@ else if ("pending".equals(status)) {
 					importedIssue.setFailure("No URL for imported issue: " + body);
 					return false;
 				}
+				var jiraResolution = jiraIssue != null ? jiraIssue.getFields().getResolution() : null;
+				if (jiraResolution != null && RESOLUTION_TYPES_FOR_NOT_PLANNED_MAPPING.contains(jiraResolution.getName())) {
```

Comment:
maybe: extract method `shouldUpdateStateReasonToNotPlanned`?  This if seems hard to read if you don't know why it exists.